src/main/java/com/company/array/RepeatedDNASequences.java

```diff
@@ -0,0 +1,55 @@
+package com.company.array;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class RepeatedDNASequences {
+    public List<String> findRepeatedDnaSequences(String s) {
+        List<String> result = new ArrayList<>();
+        if (s.length() < 11) {
+            return result;
+        }
+        Map<String, Integer> map = new HashMap<>();
+        StringBuilder str = new StringBuilder(s.substring(0, 10));
+        map.put(str.toString(), 1);
+        for (int r = 10; r < s.length(); r++) {
+            str.deleteCharAt(0);
+            str.append(s.charAt(r));
+            map.put(str.toString(), map.getOrDefault(str.toString(), 0) + 1);
+        }
+        for (Map.Entry<String, Integer> entry : map.entrySet()) {
+            if (entry.getValue() > 1) {
```

Comment:
All the values in the hashmap have a value greater than one.

Reply:
some will have 1 and some will greater than 1 and keys whose value is greater than 1 will be added to result list